index.bs

```diff
@@ -78,4 +278,322 @@ interface StorageBucket {
+
+1. Let |p| be [=a new promise=].
+
+1. Otherwise, run the following steps [=in parallel=]:
+
+    1. Let |quota| be [=/storage quota=] for |shelf|.
+
+    1. Set |quota| to |bucket|'s [=StorageBucket/quota value=] if it is non-null.
+
+    1. Let |usage| be [=storage bucket/storage usage=] for |bucket|.
+
+    1. Let |dictionary| be a new {{StorageEstimate}} dictionary whose {{StorageEstimate/usage}} member is |usage| and {{StorageEstimate/quota}} member is |quota|.
+
+    1. [=/Resolve=] |p| with |dictionary|.
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The [=StorageBucket/quota value=] will be ignored if it exceeds the total amount of space available to the [=/storage shelf=], i.e. site.
+Its intended use is to keep a specific bucket from using up the entire site's storage space.
+
+</aside>
+
+<h3 id="storage-bucket-durability">Durability</h3>
+
+A [=/storage bucket=] has a <dfn for=StorageBucket>durability value</dfn>, a {{StorageBucketDurability}}.
+The user agent MAY initialize this value to "{{StorageBucketDurability/strict}}" or
+"{{StorageBucketDurability/relaxed}}". The user agent SHOULD ignore the [=StorageBucket/durability value=]
+for operations that otherwise specify durability behavior.
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">durability()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+1. If |bucket|'s [=storage bucket/removed=] flag is true, [=reject=] |p| with an {{InvalidStateError}}.
+
+1. Otherwise, [=/resolve=] |p| with |bucket|'s [=StorageBucket/durability value=].
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The durability reflects whether the user agent will prioritize performance or durability when completing operations on data in the bucket.
+Some operations, such as {{IDBTransaction}}s, can specify a durability which will override the [=StorageBucket/durability value=]. The default
+[=StorageBucket/durability value=] is left to the user agent. Durability can be specified when a bucket is initially created, but not changed later.
+The {{StorageBucket/durability()}} method should report what the user agent will actually do. The value is one of the following:
+
+: "{{StorageBucketDurability/strict}}"
+:: The user agent will consider that operations on data in the bucket are successful only after verifying that outstanding changes have been written to a persistent storage medium.
+: "{{StorageBucketDurability/relaxed}}"
+:: The user agent will consider that operations on data in the bucket are successful as soon as all outstanding changes have been written to the operating system, without subsequent verification.
+
+</aside>
+
+<h3 id="storage-bucket-expiration">Expiration</h3>
+
+A [=/storage bucket=] has an <dfn for="StorageBucket">expiration value</dfn>, which is either null or a [=duration=], initially null.
```

Comment:
I don't think the spec is written with this being a duration currently.  It's directly comparing it to the current wall time, which suggests its a moment instead.

Reply:
Updated to moment